test/jdk/java/awt/event/ComponentEvent/ComponentEventTest.java

```diff
@@ -0,0 +1,325 @@
+                System.out.println("ComponentMoved: " + e.getSource());
+                componentMoved = true;
+                events.add(e);
+            }
+
+            @Override
+            public void componentHidden(ComponentEvent e) {
+                System.out.println("ComponentHidden: " + e.getSource());
+                componentHidden = true;
+                events.add(e);
+            }
+        };
+
+    private static void initializeGUI() {
+        frame = new Frame("Component Event Test");
+        frame.setLayout(new FlowLayout());
+
+        Panel panel = new Panel();
+        Button button = new Button("Button");
+        Label label = new Label("Label");
+        List list = new List();
+        list.add("One");
+        list.add("Two");
+        list.add("Three");
+        Choice choice = new Choice();
+        choice.add("Red");
+        choice.add("Orange");
+        choice.add("Yellow");
+        Checkbox checkbox = new Checkbox("Checkbox");
+        Scrollbar scrollbar = new Scrollbar(Scrollbar.HORIZONTAL, 0, 1, 0, 255);
+        TextField textfield = new TextField(15);
+        TextArea textarea = new TextArea(5, 15);
+
+        components = new Component[] { panel, button, label, list, choice,
+            checkbox, scrollbar, textfield, textarea, frame };
+
+        for (int i = 0; i < components.length - 1; i++) {
+            components[i].addComponentListener(componentListener);
+            frame.add(components[i]);
+        }
+        frame.addComponentListener(componentListener);
+
+        frame.pack();
+        frame.setLocationRelativeTo(null);
+        frame.setVisible(true);
+    }
+
+    public static void main(String[] args) throws Exception {
+        try {
+            robot = new Robot();
+            robot.setAutoWaitForIdle(true);
+
+            EventQueue.invokeAndWait(ComponentEventTest::initializeGUI);
+            robot.waitForIdle();
+            robot.delay(DELAY);
+
+            doTest();
+
+            System.out.println("Test PASSED");
+        } finally {
+            EventQueue.invokeAndWait(ComponentEventTest::disposeFrame);
+        }
+    }
+
+    private static void doTest()
+        throws InvocationTargetException, InterruptedException {
+
+        EventQueue.invokeAndWait(() -> {
+            compAt = components[0].getLocationOnScreen();
+            compSize = components[0].getSize();
+        });
+
+        robot.mouseMove(compAt.x + compSize.width / 2,
+            compAt.y + compSize.height / 2);
+
+        robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
+        robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
+
+        for (int i = 0; i < components.length; i++) {
+            for (boolean state : new boolean[] { true, false }) {
+                doTest(i, state);
+            }
+        }
+
+        robot.delay(DELAY);
+
+        resetValues();
+        EventQueue.invokeAndWait(() -> {
+            frame.setExtendedState(Frame.ICONIFIED);
+        });
```

Comment:
```suggestion
        EventQueue.invokeAndWait(() -> frame.setExtendedState(Frame.ICONIFIED));
```
You can use lambda expression instead of lambda statement.